pkg/action/build-iso.go

```diff
@@ -142,23 +136,51 @@ func (b BuildISOAction) prepareISORoot(isoDir string, rootDir string, uefiDir st
 	}
 
 	b.cfg.Logger.Info("Creating EFI image...")
-	err = b.createEFI(uefiDir, filepath.Join(isoDir, constants.IsoEFIPath))
+	err = b.createEFI(rootDir, isoDir)
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
-func (b BuildISOAction) createEFI(root string, img string) error {
-	efiSize, err := utils.DirSize(b.cfg.Fs, root)
+// createEFI creates the EFI image that is used for booting
+// it searches the rootfs for the shim/grub.efi file and copies it into a directory with the proper EFI structure
+// then it generates a grub.cfg that chainloads into the grub.cfg of the livecd (which is the normal livecd grub config from luet packages)
+// then it calculates the size of the EFI image based on the files copied and creates the image
+func (b BuildISOAction) createEFI(rootdir string, isoDir string) error {
+	var err error
+
+	// rootfs /efi dir
+	img := filepath.Join(isoDir, constants.IsoEFIPath)
+	temp, _ := utils.TempDir(b.cfg.Fs, "", "enki-iso")
+	_ = utils.MkdirAll(b.cfg.Fs, filepath.Join(temp, constants.EfiBootPath), constants.DirPerm)
+	_ = utils.MkdirAll(b.cfg.Fs, filepath.Join(isoDir, constants.EfiBootPath), constants.DirPerm)
```

Comment:
I guess we don't check this in case the dir is there, but what if there's an issue creating the dir?

Reply:
was already like that, just added the underscore to disable the squiggly lines in my IDE LOL
Indeed this should be checked for errors. on it